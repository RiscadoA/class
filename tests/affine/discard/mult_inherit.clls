proc main() {
    cut {
        println("affine 1"); affine x;
        println("send 1"); send x(a. println("wait 11"); wait a; ());
        println("close 1"); close x
        |x: ~affine send wait; close|
        println("affine 2"); affine y;
        println("use 1"); use x;
        println("recv 1"); recv x(a);
        println("wait 1"); wait x;
        println("send 2"); send y(a);
        println("close 2"); close y
        |y: ~affine send wait; close|
        println("discard"); discard y
    }
};;
