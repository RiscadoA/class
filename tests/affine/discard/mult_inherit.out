affine 1
affine 2
discard
